relay-general/src/store/transactions.rs

```diff
@@ -153,9 +152,7 @@ impl Processor for TransactionsProcessor {
             ));
         }
 
-        if span.op.value().is_none() {
-            return Err(ProcessingAction::InvalidTransaction("span is missing op"));
-        }
```

Comment:
All this got changed because we need a mutable reference to set `op`.

While refactoring over video with Jan, we were able to:
- Extract a common error into a variable;
- Reduce indentation levels by using multiple `let ... match` for traversing the data step by step.